docs/apis/streaming/index.md

```diff
@@ -1768,6 +1768,11 @@ greater than 1, the output will also be prepended with the identifier of the tas
 </div>
 </div>
 
+Note that the `write*()` methods on `DataStream` are mainly intended for debugging purposes.
+They are not participating in Flink's checkpointing (no fault tolerance guarantees). The 
```

Comment:
May be worth adding that this means usually "at-least-once" , but may also mean data loss in cases where the output formats buffer data and do not immediately persist it.